sw/device/tests/sensor_ctrl_alerts.c

```diff
@@ -134,6 +125,28 @@ static void test_event(uint32_t idx, dif_toggle_t fatal) {
   check_alert_state(fatal);
 };
 
+enum {
+  // Counter for event index.
+  kCounterEventIdx,
+  // Counter for number of events tested.
+  kCounterNumTests,
+  // Max number of events to test per run.
+  kNumTestsMax = SENSOR_CTRL_PARAM_NUM_ALERT_EVENTS >> 1,
+};
+
+static uint32_t get_next_event_to_test(void) {
+  uint32_t event_idx;
+  // Reseed so that we don't see the same sequence after each reset.
+  rand_testutils_reseed();
```

Comment:
i'm really confused by this one. 
Doesn't the condition mean that if we get a random number greater than half of `UINT32_MAX`, we'll just immediately terminate? 

so it seems like we would always get a result that is continuous ones and zeros right? like 11110000?
i was hoping for a more random distribution to check for any dependency between events. so something like 01001100.

Reply:
I'm copying the function here because I need the loop to explain the idea:

```
static uint32_t get_next_event_to_test(void) {
  uint32_t event_idx;
  // Reseed so that we don't see the same sequence after each reset.
  rand_testutils_reseed();
  do {
    event_idx = flash_ctrl_testutils_counter_get(kCounterEventIdx);
    flash_ctrl_testutils_counter_increment(&flash_ctrl, kCounterEventIdx);
    // Drop each event randomly to reduce run time.
  } while (rand_testutils_gen32() <= UINT32_MAX >> 1 &&
           event_idx < SENSOR_CTRL_PARAM_NUM_ALERT_EVENTS);
  return event_idx;
}
```

* The non-volatile counter starts from zero and is incremented at each iteration, i.e. `event_idx` inside the loop is `0, 1, 2, 3, 4, 5...`
* At each iteration, we either _skip_ the current event `event_idx` with 0.5 prob (assuming uniform distribution) or return it, which makes it the next event to be tested.
* When this function is called again, we resume from `event_idx + 1` thanks to the non-volatile counter.
* This continues until either we reach the last event or the counter is saturated.

So, this function actually tries to achieve what you described. This doesn't work that well on the FPGA since everything is deterministic but should work better in simulations with different seeds.

Reply:
ooooo....i think i get it now.
I was originally thinking you were trying to set the whole vector at once like the previous iteration.
I guess instead it's every time you invoke `next_event`, you'll do some randomization to determine whether the next index is returned or skipped, is that the idea?

Reply:
Exactly!

Reply:
should we make this part of the testutil? `random` increment? :)
thanks for doing this and explaining.

Reply:
Why not? captured in #14987